pkg/volume/projected/projected.go

```diff
@@ -328,6 +328,13 @@ func (s *projectedVolumeMounter) collectData() (map[string]volumeutil.FileProjec
 			}
 			tp := source.ServiceAccountToken
 
+			// When FsGroup is set, we depend on SetVolumeOwnership to
+			// change from 0600 to 0640.
+			mode := *s.source.DefaultMode
+			if mounterArgs.FsUser != nil || mounterArgs.FsGroup != nil {
+				mode = 0600
+			}
+
```

Comment:
So one flip side of this implementation is - it won't easily work for other ephemeral volume types, which was discussed in sig-storage call.

Reply:
yea, it is intended to limit the scope to only projected service account token but this can be easily extended to other projected sources, non-projected secrets, non-projected configMap..etc.